t/t1005-max-jobs-limits.t

```diff
@@ -92,12 +92,16 @@ test_expect_success 'submit max number of jobs' '
 
 test_expect_success 'submit job while already having max number of running jobs' '
 	jobid3=$(flux python ${SUBMIT_AS} 5011 sleep 60) &&
-	test $(flux jobs -no {state} ${jobid3}) = DEPEND
+	flux job wait-event -vt 60 $jobid3 dependency-add &&
+	flux job info $jobid3 eventlog > eventlog.out &&
+	grep "dependency-add" eventlog.out
```

Comment:
The `validate` event should transition a job from NEW to DEPEND, so it may be possible to replace the test for `state == DEPEND` with `flux job wait-event JOBID validate`. However, I may be misunderstanding the intent here.

Also, it probably isn't necessary to use `flux job wait-event` and then grep for the event in the eventlog fetched from `flux job info`. In this case, if `flux job wait-event` has reported that the event has appeared, it is guaranteed that the event is posted to the eventlog. Therefore I _think_ you could skip the two extra steps here as redundant.

Reply:
Thanks for this pointer! The intent with this test is to ensure that the job has a dependency added to it by the priority plugin. In this case, should I still change the check to instead use your suggestion `flux job wait-event JOBID validate`? I should update the description of the test to make it clearer what it is trying to do, sorry about that.

Thanks also for your suggestion about adding `flux job info` after a `flux job wait-event` - I can just remove those from this test and the others that have it.

Reply:
Sorry, I misunderstood the intent. If the intent is to ensure a dependency is added, the `flux job wait-event dependency-add` is the correct approach.

You could also consider using the `flux job wait-event` `-m, --match-context` option to ensure that the context matches the dependency you expect to be adding too. 

Sorry for my confusion!

Reply:
Not a problem! I just force-pushed up a change to use `--match-context` with `flux job wait-event` per your suggestion above (I also updated the test description to hopefully be a little bit clearer), as well as removed the redundant `flux job info` calls throughout the changed tests that look at job state.